app/imaging/utils.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package imaging
+
+import (
+	"image"
+	"image/color"
+
+	"github.com/disintegration/imaging"
+)
+
+type rawImg interface {
+	Set(x, y int, c color.Color)
+	Opaque() bool
+}
+
+func isFullyTransparent(c color.Color) bool {
+	var a uint32
+	switch c.(type) {
```

Comment:
Why this type switch is needed?

Reply:
As is not at much. The idea there was to possibly optimize further by knowing the type of color and extract only what's needed (the alpha value). Now it's just defaulting to the sub-optimal case.

Reply:
Simplified and added a TODO.